proto/bpfd.proto

```diff
@@ -75,59 +139,88 @@ message LoadRequest {
     }
 };
 
-message LoadResponse {
-    uint32 id = 1;
+/* LoadRequest represents a request to load and attach a bpf program. */
+
+message LoadRequest {
+    BytecodeLocation bytecode = 1;
+    string name = 2;
+    uint32 program_type = 3;
+    AttachInfo attach = 4;
+    map<string,string> metadata = 5;
+    map<string, bytes> global_data = 6;
+    optional string uuid = 7;
+    optional uint32 map_owner_id = 8;
+};
+
+/* LoadResponse represents a response from loading and attaching an eBPF program. 
+ * It includes all of the state kept by bpfd and the Linux kernel for the
+ * program.
+ */
+ 
+ message LoadResponse {
+    ProgramInfo info  = 1;
+    KernelProgramInfo kernel_info = 2;
 }
 
+/* UnloadRequest represents a request to unload an eBPF program that was loaded
+ * and attached by bpfd. To identify a program pass in a valid kernel ID.
+ */
+
 message UnloadRequest {
     uint32 id = 1;
 }
 
 message UnloadResponse {}
 
+/* ListRequest represents a request to get information regarding eBPF programs
+ * that are loaded and attached by bpfd AND/OR programs that are loaded by other
+ * users.
+ */
+
 message ListRequest {
     optional uint32 program_type = 1;
     optional bool bpfd_programs_only = 2;
     map<string, string> match_metadata = 3;
 }
 
+/* ListResponse represents a response a request to list loaded and attached
+ * eBPF programs. Based on the corresponding ListRequest programs will include
+ * those loaded by bpfd AND/OR those loaded by other users. The ListResults
+ * will ALWAYS contain a program's kernel info and if loaded by bpfd, will
+ * also contain the state kept by bpfd.
+ */
+
 message ListResponse {
-  message ListResult {
-    uint32 id = 1;
-    string name = 2;
-    oneof location {
-        BytecodeImage image = 3;
-        string file = 4;
-    } ;
-    uint32 program_type = 5;
-    map<string, bytes> global_data = 6;
-    optional uint32 map_owner_id = 7;
-    string map_pin_path = 8;
-    repeated string map_used_by = 9;
-    oneof attach_info {
-        XDPAttachInfo xdp_attach_info = 10;
-        TCAttachInfo tc_attach_info = 11;
-        TracepointAttachInfo tracepoint_attach_info = 12;
-        KprobeAttachInfo kprobe_attach_info = 13;
-        UprobeAttachInfo uprobe_attach_info = 14;
-    }
-    map<string, string> metadata = 15;
-    string loaded_at = 16;
-    string tag = 17;
-    bool gpl_compatible = 18;
-    repeated uint32 map_ids = 19;
-    uint32 btf_id = 20;
-    uint32 bytes_xlated = 21;
-    bool jited = 22;
-    uint32 bytes_jited = 23;
-    uint32 bytes_memlock = 24;
-    uint32 verified_insns = 25;
-  } 
-  repeated ListResult results = 26;
+  message ListResult { 
+    ProgramInfo info = 1;
```

Comment:
`ProgramInfo` should be optional here I think since not every program will have bpfd specific informaton

Reply:
Done